### PR TITLE
use the cni test image that we've just built for our tests

### PR DESCRIPTION
**Pull request by stevej**
We were using an edge release before which was not testing our changes.

Signed-off-by: Steve Jenson <stevej@buoyant.io>

### CODE REVIEW

File: cni-plugin/main.go [hunk 1]
```diff
@@ -134,9 +133,10 @@ func cmdAdd(args *skel.CmdArgs) error {
 	logrus.Debug("linkerd-cni: cmdAdd, parsing config")
 	conf, err := parseConfig(args.StdinData)
 	if err != nil {
+		logrus.Errorf("error parsing config: %e", err)
 		return err
 	}
-	configureLogging(conf.LogLevel)
+	configureLoggingLevel(conf.LogLevel)
```
Comment (mateiidavid) [inline cni-plugin/main.go:139]: I know it's not part of this change, but I never understood why we logged on L133 and then configured the logging a few lines below it. Surely we should just configure it and then log? Would it make sense to just move this to the top of the function's scope now?

File: internal/iptables/iptables.go [hunk 2]
```diff
@@ -234,7 +234,9 @@ func executeCommand(firewallConfiguration FirewallConfiguration, cmd *exec.Cmd)
 		// command.
 		//
 		// See https://github.com/rancher/k3s/issues/1434#issuecomment-629315909
-		args := append([]string{"--net", firewallConfiguration.NetNs, "--"}, cmd.Args...)
+		args := []string{fmt.Sprintf("--net=%s", firewallConfiguration.NetNs)}
+		args = append(args, "--")
+		args = append(args, cmd.Args...)
```
Comment (mateiidavid) [inline internal/iptables/iptables.go:239]:
```go
nsArgs := fmt.Sprintf("--net=%s", firewallConfiguration.NetNs)
args := []string{nsArgs, "--", cmd.Args...)
```

An alternative way to writing this that is a bit more concise. While it probably won't do much in terms of performance (we're working with small data here) I think it's nice to have the array already initialised.

I trust your judgement more than mine so I'll leave it up to you. TIOLI :D

**Review by mateiidavid — APPROVED** [review]
Nice! 🚀 Almooost there. Feels great to finally be able to test all of this, and feels great to bump the proxy-init library. Can't believe we haven't been on the most recent version for what, 5 months now?

I left a few non-blocking comments for your consideration. Ship whenever.

File: cni-plugin/main.go (hunk 1 above)
Comment (stevej) [reply cni-plugin/main.go:139]: There's a chicken-and-egg situation here, you can't change the logging level until you parse the config and the debug log message on L133 exists to give us a hint that we're about to parse the config. We can remove the whole line, in my opinion.

File: internal/iptables/iptables.go (hunk 2 above)
Comment (stevej) [reply internal/iptables/iptables.go:239]: Is there another syntax for this?

```
internal/iptables/iptables.go:241:42: missing ',' in composite literal (typecheck)
                args := []string{nsArgs, "--", cmd.Args...}
```

Comment (mateiidavid) [reply internal/iptables/iptables.go:239]: Oh I think I misread the code a little bit and gave a bad suggestion:

```
args := append([]string{nsArgs, "--"}, cmd.Args...)
```

this is what I had in mind

File: cni-plugin/main.go (hunk 1 above)
Comment (mateiidavid) [reply cni-plugin/main.go:139]: Yeah, let's just remove it maybe.

File: internal/iptables/iptables.go (hunk 2 above)
Comment (stevej) [reply internal/iptables/iptables.go]: Thanks! I resolved this and the question about the debug log line.

**Pull request merged**